Waterfall settings: color settings for increases, decreases, total

ariya opened the pull request:
**Steps to test**

1. Ask a question, Native query
2. Choose Sample Dataset
3. Type the following and then Run query (Ctrl+Enter)

```sql
select 'X' as product, 5 as profit 
union all 
select 'Y' as product, -8 as profit 
union all 
select 'Z' as product, 2 as profit 
```

4. Visualization, Waterfall
5. Settings, Display

**Expected**: There should be 3 widgets to choose the color for increase, decrease, and total. Use the color picker to change the color in the chart (for increase or decrease or total).

![image](https://user-images.githubusercontent.com/7288/100945344-e32c7a80-34b5-11eb-9cfc-1b6a400f9003.png)

Commit:
Waterfall settings: color widgets for increases, decreases, total

ariya commented:
This only implements the color widgets so far, the colors still have no effect on the rendering.

@paulrosenzweig I can't seem to find a way to group the color widgets, to make it more like the series color in the regular bar chart. Any tips or ideas?

paulrosenzweig commented on frontend/src/metabase/visualizations/visualizations/WaterfallChart.jsx:
```diff
@@ -5,20 +5,38 @@ import { assocIn } from "icepick";
     ...GRAPH_AXIS_SETTINGS,
+    "waterfall.increase_color": {
+      section: t`Display`,
+      title: t`Increase color`,
```
You could make it slightly more compact by moving the title inside the widget:

```suggestion
      props: {title: t`Increase color`},
```

ariya replied:
Precisely what I need! Thank you.

paulrosenzweig reviewed (approved):
Looks good!

Commit:
Compactify the color settings for waterfall

Commit:
Waterfall colors: apply the increase & decrease colors

Commit:
Move waterfall coloring to setChartColor()

Commit:
Separate waterfall's total into its own (in the virtual stacked bars)

This makes it easy to color the total bar, which has a value only for
the last one in the series anyway

Commit:
Waterfall total color should be conditional on the total toggle

paulrosenzweig reviewed (approved):
🥳
